### PR TITLE
MTDSA-18202 - checkpoint OAS

### PR DESCRIPTION
**Pull request by jeremystone**
- add OAS feature switch Handlebars support
 - conditionalise checkpoint OAS
 - add create checkpoint OAS

### CODE REVIEW

File: test/config/DocumentationControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import org.scalatest.OptionValues
+import play.api.Configuration
+import play.api.http.{DefaultFileMimeTypes, DefaultHttpErrorHandler, FileMimeTypesConfiguration, HttpConfiguration}
+import play.api.mvc.Result
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class DocumentationControllerSpec extends ControllerBaseSpec with MockAppConfig with OptionValues {
```
Comment (harry-ritchie) [inline test/config/DocumentationControllerSpec.scala:34]: Is it also worth unit testing the `definition` method in the controller?

File: app/config/rewriters/DocumentationRewriters.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+object DocumentationRewriters {
+
+  trait CheckRewrite {
+    def apply(version: String, filename: String): Boolean
+  }
+
+  case class CheckAndRewrite(check: CheckRewrite, rewrite: Rewriter) {
+
+    def maybeRewriter(version: String, filename: String): Option[Rewriter] =
```
Comment (harry-ritchie) [inline app/config/rewriters/DocumentationRewriters.scala:41]: Is it worth unit testing this exposed method?

File: resources/public/api/conf/1.0/checkpoints_create.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+summary: Create a checkpoint
+description: |
+  This endpoint allows a developer to set up a checkpoint for the stateful data in the 
+  sandbox environment for a particular NINO. A National insurance number must be provided.
```
Comment (harry-ritchie) [inline resources/public/api/conf/1.0/checkpoints_create.yaml:4]: Minor wording thing, but should we be consistent with using the phrase `NINO` or `National insurance number` (does referencing them different in the same description imply they are different?)?

Comment (jeremystone) [reply resources/public/api/conf/1.0/checkpoints_create.yaml:4]: Probably should use "National insurance number". I will fix.

File: test/config/DocumentationControllerSpec.scala (hunk 1 above)
Comment (jeremystone) [reply test/config/DocumentationControllerSpec.scala:34]: It is tested via the ITs so I don't think so (and it is a pretty simple method).

File: app/config/rewriters/DocumentationRewriters.scala (hunk 2 above)
Comment (jeremystone) [reply app/config/rewriters/DocumentationRewriters.scala:41]: Possibly, but again it is a very simply method (and it is covered by the ITs), and there are no UTs for it in the SE business API where I found it.

**Review by harry-ritchie — APPROVED** [review]

**Pull request merged**